pulumi-language-dotnet/main.go

```diff
@@ -559,17 +559,28 @@ func (w *logWriter) LogToUser(val string) (int, error) {
 	return len(val), nil
 }
 
-func (host *dotnetLanguageHost) buildDll() (string, error) {
+func (host *dotnetLanguageHost) buildDll(entryPoint string) (string, error) {
 	// If we are running from source, we need to build the project.
 	// Run the `dotnet build` command.  Importantly, report the output of this to the user
 	// (ephemerally) as it is happening so they're aware of what's going on and can see the progress
 	// of things.
 	args := []string{"build", "-nologo", "-o", "bin/pulumi-debugging"}
+	if entryPoint != "" {
+		args = append(args, entryPoint)
+	}
 
 	cmd := exec.Command(host.exec, args...)
-	err := cmd.Run()
+	out, err := cmd.CombinedOutput()
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to build project: %v", err)
+		return "", errors.Wrapf(err, "failed to build project: %v, output: %v", err, string(out))
+	}
+
+	if entryPoint != "" {
+		lastDot := strings.LastIndex(entryPoint, ".")
+		if lastDot == -1 {
+			return "", errors.New("entry point must have a file extension")
+		}
+		return filepath.Join("bin", "pulumi-debugging", entryPoint[:lastDot]+".dll"), nil
```

Comment:
Looking at the code, it feels like `buildDll` (`buildDLL`? surprised the linter let that one slip) is only involved in debugging codepaths. If so, should we rename it to `buildDebuggingDLL` or the like? And perhaps a doc comment to explain what's going on/take some of the great stuff from the PR into this context?

Reply:
Done both of these. thanks!  

I'm not even sure we run the go linter in this repo (another reason to merge this back into pulumi/pulumi :sweat_smile:)